crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -302,6 +321,66 @@ impl C8yMapperActor {
 
         Ok(())
     }
+
+    async fn start_initial_heartbeat_timer(&mut self) -> Result<(), RuntimeError> {
+        if let Some(interval) = self.converter.get_availability_period_if_enabled() {
+            let topic_id = &self.converter.device_topic_id;
+            let health_topic = default_main_lead_service_topic(topic_id);
+
+            set_heartbeat_timer(
+                interval,
+                &mut self.converter.health_status,
+                &mut self.converter.active_heartbeat_timer,
+                topic_id,
+                health_topic,
+                self.timer_sender.clone(),
+            )
+            .await?;
+        }
+
+        Ok(())
+    }
+
+    async fn process_heartbeat(
+        &mut self,
+        heartbeat_payload: HeartbeatPayload,
+    ) -> Result<(), RuntimeError> {
+        match self
+            .converter
+            .active_heartbeat_timer
+            .get(&heartbeat_payload.device)
+        {
+            Some(health_topic) if health_topic == &heartbeat_payload.health => {
+                if let Some(interval) = self.converter.get_availability_period_if_enabled() {
+                    if let Ok(inventory_update_topic) = self
+                        .converter
+                        .get_inventory_update_topic(&heartbeat_payload.device)
+                    {
+                        if let Some(message) = create_c8y_heartbeat_message(
+                            &self.converter.health_status,
+                            &inventory_update_topic,
+                            &heartbeat_payload,
+                        ) {
+                            self.mqtt_publisher.send(message).await?;
+                        }
+                    }
+
+                    start_heartbeat_timer(
+                        self.timer_sender.clone(),
+                        interval,
```

Comment:
Using the same interval for the timer that we've set as the `c8y_RequiredAvailability` interval would lead to a lot of missed heartbeats. For e.g, if the interval is set to 30 mins and if we sleep for 30 minutes here using the timer, by the time we process the heartbeat response from the timer and send a response to the cloud, it definitely would take more than 30 mins, by then the availability interval in the cloud would have timed out. So, IMO, we should send heartbeat intervals more often than what the configured values is.